modules/oe_theme_helper/src/Plugin/field_group/FieldGroupFormatter/FieldInPageNavigationFormatter.php

```diff
@@ -0,0 +1,188 @@
+        continue;
+      }
+      $type = $this->groupOrField($value, $processed_object['#fieldgroups']);
+      // If a group has a default id, use it as anchor link.
+      if ($type === 'group' && !empty($processed_object['#fieldgroups'][$value]->format_settings['id'])) {
+        $id = $processed_object['#fieldgroups'][$value]->format_settings['id'];
+      }
+      else {
+        $id = "inline-nav-" . $i = isset($i) ? ++$i : 1;
+      }
+
+      $links[] = [
+        'href' => "#" . $id,
+        'label' => $label,
+      ];
+      $element[$value]['#id'] = $id;
+      $element[$value]['#attributes']['id'] = $id;
+    }
+
+    $element['#links'] = $links;
+    if (count($links) > 0) {
+      $element['#inline_title'] = t('Page contents');
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function preRender(&$element, $rendering_object) {
+    parent::preRender($element, $rendering_object);
+    $this->process($element, $rendering_object);
+  }
+
+  /**
+   * Check if we should exclude this field from the in-page navigation menu.
+   *
+   * @param string $field_name
+   *   Name of the field.
+   * @param array $field
+   *   The field to check.
+   * @param array $field_groups
+   *   List of groups.
+   *
+   * @return bool
+   *   Exclude or not the field.
+   */
+  protected function excludeField(string $field_name, array $field, array $field_groups) :bool {
+    $type = $this->groupOrField($field_name, $field_groups);
+    switch ($type) {
+      case "field":
+        // Exclude empty fields.
+        if (!isset($field['#items'])) {
+          return TRUE;
+        }
+
+        break;
+
+      case 'group':
+        // Exclude empty groups.
+        if (count($field) < 1) {
+          return TRUE;
+        }
+
+        foreach ($field as $value) {
+          // If there is a field with values, don't exclude the group.
+          if (isset($value['#items'])) {
+            return FALSE;
+          }
+        }
+
+        // Exclude groups with only empty fields.
+        return TRUE;
+    }
+
+    return FALSE;
+  }
+
+  /**
+   * Get the label used on the field.
+   *
+   * @param string $field_name
+   *   Name of the field.
+   * @param array $processed_object
+   *   Entity processed.
+   *
+   * @return string
+   *   Label of the field.
+   */
+  protected function fieldGetLabel(string $field_name, array $processed_object): string {
+    $label = '';
+    $type = $this->groupOrField($field_name, $processed_object['#fieldgroups']);
+    switch ($type) {
+      case "field":
+        // Show the label if it is not hidden.
+        if (!in_array($processed_object[$field_name]['#label_display'], ['hidden', 'visually_hidden'])) {
+          $title = $processed_object[$field_name]['#title'];
+          if (is_string($title)) {
+            $label = $processed_object[$field_name]['#title'];
+          }
+          if (is_object($title) && method_exists($title, "__toString")) {
+            $label = $title->__toString();
```

Comment:
You don't need to do this, if objects labels are renderable as strings.

Reply:
Removed